ScoutSuite/output/data/inc-scoutsuite/pagination.js

```diff
@@ -0,0 +1,162 @@
+var defaultPageSize = 2
+var regionPageSize = 1
+var reCount = new RegExp('_count+$')
+
+/**
+ * Loads a page based on which page we want to move to
+ * @param {array} pathArray         The path of where the data is stored
+ * @param {number} indexDiff        Difference between current and desired page index
+ */
+function loadPage (pathArray, indexDiff) {
+  let pageInfo = getPageInfo(pathArray)
+  let pageSize = pageInfo[0]
+  let pageIndex = pageInfo[1]
+  pageIndex += indexDiff
+  // getResourcePageSqlite being called in both statements is intentional, I want events to happen in this order to
+  // prevent the user from cliking on next page multiple times and going out of bounds and I want to call loadConfig
+  // to regenerate the page after Iv'e loaded SQLite data
+  if (indexDiff === 0) {
+    if (pathArray.length === 3) {
+      getResourcePageSqlite(pageIndex, pageSize, pathArray[1], pathArray[2])
+    } else if (pathArray.length === 5) {
```

Comment:
I'm a bit confused by the pagination logic. It seems like there is AWS-specific applied to all the providers (I'm referring to the regions). I'm a bit confused by the `pathArray.length === 3` and `pathArray.length === 5`. I'm not sure where the numbers come from, and I feel like it could easily break when introducing new providers.

Reply:
Ok, I noticed the think which checks if the path has regions. Is there a way to make the pagination system more generic so we don't have to have provider-specific logic?

Reply:
When there are no regions the path looks like `services.[service].[resource]` (length of 3) while when there are regions it looks like  `services.[service].regions.[region].[resource]` (length of 5). Atm we only have AWS that uses regions, but not all services of AWS use regions.

I'll change it to verify whether the third element is `regions` or not then.